Docs/usman plugin mgmt (#85424)
grafana/grafana: docs/sources/administration/plugin-management/index.md

labels:
  products:
    - enterprise
    - cloud
    - oss
title: Plugin management
weight: 600
---

# Plugin management

You can enhance your Grafana experience with _plugins_, extensions to Grafana beyond the wide range of visualizations and data sources that are built-in.

This guide shows you how to [install](#install-a-plugin) plugins that are built by Grafana Labs, commercial partners, our community, or plugins that you have [built yourself](/developers/plugin-tools).

## Types of plugins

Grafana supports three types of plugins:

- [Panels](/grafana/plugins/panel-plugins) - These plugins make it easy to create and add any kind of panel, to show your data, or improve your favorite dashboards.

> - [Data sources](/grafana/plugins/data-source-plugins) - These plugins allow you to pull data from various data sources such as databases, APIs, log files, etc and display it in the form of graphs, charts, and dashboards in Grafana.

Reviewer: ```suggestion
- [Data sources](/grafana/plugins/data-source-plugins) - These plugins allow you to pull data from various data sources such as databases, APIs, log files, and so on, and display it in the form of graphs, charts, and dashboards in Grafana.
```

# Plugin management

You can enhance your Grafana experience with _plugins_, extensions to Grafana beyond the wide range of visualizations and data sources that are built-in.

This guide shows you how to [install](#install-a-plugin) a wide range of the plugins that are built by Grafana Labs, commercial partners, our community, or plugins that you have [built yourself](/developers/plugin-tools).

## Types of plugins

Grafana supports three types of plugins:

> - [Panels](/grafana/plugins/panel-plugins) - These plugins make it easy to create and add any kind of panel, to show your data, or improve your favorite dashboards.

Author: @sympatheticmoose : How about:

- [Panels](/grafana/plugins/panel-plugins) - also known as visualizations, allows Grafana to use custom visualizations and display it in a Dashboard?

Besides the wide range of visualizations and data sources that are available immediately after you install Grafana, you can extend your Grafana experience with _plugins_.

> You can [install](#install-a-plugin) a wide range of the plugins that are built by the Grafana Labs, Partners, Community, or [build one yourself](/developers/plugin-tools).

Reviewer 2: ```suggestion
You can [install](#install-a-plugin) a wide range of the plugins that are built by Grafana Labs, commercial partners, our community, or [build one yourself](/developers/plugin-tools).
```

Grafana supports three types of plugins:

> - [Panels](/grafana/plugins?type=panel)
> - [Data sources](/plugins?type=datasource)
> - [Apps](/grafana/plugins?type=app)

Reviewer 2: ```suggestion
- [Panels](/grafana/plugins/panel-plugins)
- [Data sources](/grafana/plugins/data-source-plugins)
- [Apps](/grafana/plugins/app-plugins)
```

Its odd that they were different, and also redirects so maybe better to go with where they actually land - (please verify this with a build of the docs)

## Data source plugins

> Data source plugins add support for new databases, such as [Google BigQuery](/grafana/plugins/grafana-bigquery-datasource), [Zendesk](https://grafana.com/grafana/plugins/gabrielthomasjacobs-zendesk-datasource/) and [NetData](https://grafana.com/grafana/plugins/netdatacloud-netdata-datasource/) Data source plugins.

Reviewer 2: Required change: Zendesk and NetData aren't what I would think of as databases, and bookending the start and end with "Data source plugins" feels strange to me.

Author: The reason I mentioned them is that I want to share the combination of Data sources which are owned by Grafana (Zabbix), Community (Zendesk) and Partner (Netdata). 

If you suggest that it may not be the required then I can simply undo my changes here.

Data source plugins communicate with external sources of data and return the data in a format that Grafana understands. By adding a data source plugin, you can immediately use the data in any of your existing dashboards.

> Use data source plugins when you want to import data from external or 3rd party systems.

Reviewer 2: This is a slightly strange sentence given data source plugins are also how we query things like loki whether thats in the same Grafana stack in cloud or self hosted.

Author: I actually took it from https://grafana.com/developers/plugin-tools/introduction/plugin-types-usage#usage-of-data-source-plugins where it says "when you want to work with data provided by a third-party service".

## App plugins

> Applications, or _app plugins_, bundle data sources and panels to provide a cohesive experience, such as the [Zabbix](/grafana/plugins/alexanderzobnin-zabbix-app), [Performance Co-Pilot](https://grafana.com/grafana/plugins/performancecopilot-pcp-app/) and [cPacket Utility Bundle](https://grafana.com/grafana/plugins/cpacket-cclearutilitybundle-app/) apps.

Reviewer 2: This is generally a bad description of apps given theres no requirement to bundle anything... which I know is outside the scope of these changes, BUT technically, the cPacket one doesn't include a data source (the others do) so unless we change the description, we should change the example.

Author: Same intention as mentioned above in Data Sources. 

Since I am not too expert with Apps so I can simply undo the changes here.

Use app plugins when you want an out-of-the-box monitoring experience.

### Managing access for app plugins

With [RBAC]({{< relref "../roles-and-permissions/access-control/#about-rbac" >}}), you can now customize access to app plugins.

By default, the roles of Viewers, Editors and Admins have access to all app plugins that their Organization role allows them to access. Access is granted by the `fixed:plugins.app:reader` role.

{{% admonition type="note" %}}

> Revoking the RBAC role from some users will prevent them from accessing app plugins. However, granting the RBAC role to users will only allow them to see app plugins their Organization role allows them to see.

Author: I agree I re-read the NOTE and 

1st: I do not understand its ending where it says "allows them to see". It feels like some more information is missing. 

2nd: Just before the NOTE it explains the default behavior and then in note about revoking but not pointing to the doc link.

So yes this NOTE does not make that much sense here and can even confuse OSS users.

## Plugin catalog

The Plugin catalog allows you to browse and manage plugins from within Grafana. Only Grafana server administrators and organization administrators can access and use the plugin catalog. For more information about Grafana Roles and Permissions, refer to [Roles and permissions]({{< relref "../administration/roles-and-permissions" >}}). The following access rules apply depending on the user role:

- If you are an **Org Admin**, you can configure app plugins but you cannot install/uninstall/update them.
- If you are a **Server Admin**, you cannot configure app plugins but you can install/uninstall/update them.

> - If you are both (**Org + Server Admin**), you can configure app plugins and also install/uninstall/update them.

Reviewer 2: I still feel like this is confusing, but this time because it implies that it is only about app plugins, whereas install/uninstall/update refers to **all** plugins. 

@josmperez perhaps you could suggest something?

I had originally suggested:
> - The **server admin** role is required to install/uninstall/update plugins.
> - The **org admin** role is required to configure app plugins.

Maybe a change to the preceding paragraph would also help

_Video shows the Plugin catalog in a previous version of Grafana._

In order to be able to install / uninstall / update plugins using plugin catalog, you must enable it via the `plugin_admin_enabled` flag in the [configuration]({{< relref "../../setup-grafana/configure-grafana/#plugin_admin_enabled" >}}) file.

{{% admonition type="note" %}}

> In the latest versions of Grafana, the `plugin_admin_enabled` flag is enabled by default.

Reviewer 2: I'd be tempted to just remove this and the previous paragraph instead.

Author: Acknowledge

### Install a plugin

To install a plugin:

1. In Grafana, click **Administration > Plugins and data > Plugins** in the side navigation menu to view all plugins.
1. Browse and find a plugin.
1. Click the plugin's logo.
1. Click **Install**.

When the update is complete, you'll see a confirmation message that the installation was successful.

### Update a plugin

To update a plugin:

1. In Grafana, click **Administration > Plugins and data > Plugins** in the side navigation menu to view all plugins.
1. Click the **Installed** filter to show only installed plugins.
1. Click the plugin's logo.
1. Click **Update**.

When the update is complete, you'll see a confirmation message that the update was successful.

### Uninstall a plugin

To uninstall a plugin:

1. In Grafana, click **Administration > Plugins and data > Plugins** in the side navigation menu to view all plugins.
1. Click the plugin's logo.
1. Click the **Installed** filter to show only installed plugins.
1. Click **Uninstall**.

When the update is complete, you'll see a confirmation message that the uninstall was successful.

## Install Grafana plugins

Grafana supports data source, panel, and app plugins.

> 1. In a web browser, navigate to the [Grafana plugin catalog](/plugins) and find a plugin that you want to install.

Reviewer: ```suggestion
1. In a web browser, navigate to the [Grafana plugin catalog](https://grafana.com/plugins) and find a plugin that you want to install.
```

_Video shows the Plugin catalog in a previous version of Grafana._

> In order to be able to install, uninstall, or update plugins using the plugin catalog, you must enable it with the `plugin_admin_enabled` flag in the [configuration]({{< relref "../../setup-grafana/configure-grafana/#plugin_admin_enabled" >}}) file.

Reviewer 2: ```suggestion
```

Author: Agree on the changes due to the fact:

1. It is already enabled by default.
2. So, far in my experience while working with users, no one come and asked as how to disable this specific part.

{{% admonition type="note" %}}

> In the most recent versions of Grafana, the `plugin_admin_enabled` flag is enabled by default.

Reviewer 2: ```suggestion
If required, the Grafana plugin catalog can be disabled using the `plugin_admin_enabled` flag in the [configuration]({{< relref "../../setup-grafana/configure-grafana/#plugin_admin_enabled" >}}) file.
```

## Install Grafana plugins

> Grafana supports data source, panel, and app plugins.

Reviewer 2: This whole paragraph feels like unnecessary repetition of content we've covered above, or should have. At this stage it is task based documentation and we shouldn't bloat it.

Author: I also agree here as this felt repetition for me too and I also agree that we should simply remove it

Reviewer: In my latest commit, I moved to the content up to the top where types were introduced without description, and removed it here. Does that improve?

### Install plugin on local Grafana

> Follow the instructions on the Install tab. You can either install the plugin with a Grafana CLI command or by downloading and uncompress a `.zip` file into the Grafana plugins directory. We recommend using Grafana CLI in most instances. The `.zip` option is available if your Grafana server does not have access to the internet.

Reviewer 2: The challenge I have is that we have just described the primary installation method, the catalog. 

As minor editorial changes, these are fine, but I feel they are not improving the page - as we should be thinking about whether the sections are required or should be reframed.

For Cloud, given the default is now that plugin installation is the same as on premise, and we have Grafana Cloud docs, this section feels pointless within this page and is more a "detail" for the cloud docs perhaps. The for local, instead this could be about "Installing plugins using the Grafana CLI"

Author: Good Point.

I suggest the following. We keep the heading https://grafana.com/docs/grafana/latest/administration/plugin-management/#install-plugin-on-local-grafana

And then mention that users can use the Plugin Catalog as it is available by default. And if want to install via CLI then refer them to the https://grafana.com/docs/grafana/latest/cli/ which itself is very detailed (I can move the unzip example there so that we do not loose it).

For the cloud do you know the page which we can refer to?

| Signature status   | Description                                                                     |
| ------------------ | ------------------------------------------------------------------------------- |
| Core               | Core plugin built into Grafana.                                                 |
| Invalid signature  | The plugin has an invalid signature.                                             |
| Modified signature | The plugin has changed since it was signed. This may indicate malicious intent. |
| Unsigned           | The plugin is not signed.                                                       |
| Signed             | The plugin signature was successfully verified.                                 |

> ### Plugin signature levels

Reviewer: In the Developer Portal, we have removed a similar section and instead link directly to the [official definitions][(https://grafana.com/legal/plugins/#what-are-the-different-classifications-of-plugins](https://grafana.com/legal/plugins/#what-are-the-different-classifications-of-plugins)) provided by Grafana Legal. This eliminates inconsistencies and improves maintainability. Would that work here?

Reviewer 2: These are different, its more about the signature check than the signature type
